compiler/src/dotty/tools/backend/jvm/ClassfileWriters.scala

```diff
@@ -182,7 +181,7 @@ class ClassfileWriters(frontendAccess: PostProcessorFrontendAccess) {
       jarWriter.putNextEntry(entry)
       try jarWriter.write(bytes, 0, bytes.length)
       finally jarWriter.flush()
-      null
+      new VirtualFile(file.absolutePath + "!" + relativePath, bytes)
```

Comment:
For a file inside a JAR we create a `VirtualFile` with path: `/path/to/file.jar!relative/path/to/file.class `

This is also the convention in Zinc: https://github.com/sbt/zinc/blob/68b9c1a7e203dd20dea40e293574c50f5b22f662/internal/compiler-bridge/src/main/scala/xsbt/JarUtils.scala#L46-L48

Reply:
So my comment here is that VirtualFile is hardcoded to return null in its `jpath` which is needed by `onFileCreated` - so I think this is the wrong abstraction - perhaps PlainFile with the name funny path